File: app/src/main/java/io/trewartha/positional/CoordinatesFragmentPagerAdapter.java
package io.trewartha.positional;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

public class CoordinatesFragmentPagerAdapter extends FragmentPagerAdapter {

    @NonNull private Fragment[] fragments;

    public CoordinatesFragmentPagerAdapter(
            @NonNull FragmentManager fragmentManager,
            @NonNull Fragment[] fragments
    ) {
        super(fragmentManager);
        this.fragments = fragments;
    }

    @Override
    public Fragment getItem(int position) {
        return fragments[position];
    }

    @Override
    public int getCount() {
        return fragments.length;
    }
}

File: app/src/main/java/io/trewartha/positional/ui/location/coordinates/CoordinatesFragmentPagerAdapter.kt
package io.trewartha.positional.ui.location.coordinates

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class CoordinatesFragmentPagerAdapter(
        fragmentManager: FragmentManager
) : FragmentPagerAdapter(fragmentManager) {

    private val fragments = arrayOf(
            CoordinatesDecimalDegreesFragment(),
            CoordinatesDegreesDecimalMinutesFragment(),
            CoordinatesDegreesMinutesSecondsFragment(),
            CoordinatesUTMFragment(),
            CoordinatesMGRSFragment()
    )

    override fun getItem(position: Int): Fragment = fragments[position]

    override fun getCount(): Int = fragments.size
}
